src/Lean/Expr.lean

```diff
@@ -459,7 +459,7 @@ inductive Expr where
   -/
   | proj (typeName : Name) (idx : Nat) (struct : Expr)
 with
-  @[computed_field, extern c inline "lean_ctor_get_uint64(#1, lean_ctor_num_objs(#1)*sizeof(void*))"]
+  @[computed_field, extern "lean_expr_data"]
```

Comment:
@Kha I am assuming you are reviewing this PR. If I am correct, could you please ensure that every "c inline" that was converted into a function has been properly marked as an "static inline"? This is the case for this one, but I didn't check all of them.

Reply:
Yes, they are neatly lined up here: https://github.com/leanprover/lean4/pull/2340/files#diff-210fa1e51e911735f3210d99d02d03b44aa715dcfb93f428885b5b830882b29c
When the PR is functioning, we should do a performance run for each backend to check for regressions